src/utils/zulipVersion.js

```diff
@@ -0,0 +1,149 @@
+/* @flow strict-local */
+
+import type { JSONable } from './jsonable';
+
+type VersionElements = {
+  major: number | void,
+  minor: number | void,
+  patch: number | void,
+  flag: 'dev' | ['rc', number] | void,
+  numCommits: number | void,
+  commitId: string | void,
+};
+
+/**
+ * Used 1) to compare a received zulipVersion from the server to a threshold with .isAtLeast.
+ *
+ * Also used 2) to report the raw server version to Sentry with .raw(), and with
+ * .loggingArray() which informs event aggregation; e.g., 1.2.3 -> [1, 2, 3].
+ *
+ * The ZulipVersion instance itself should not be stored in Redux or sent to
+ * Sentry because it isn't serializable. .raw() and .loggingArray() are marked
+ * as JSONable and can safely be used.
+ */
+export class ZulipVersion {
+  _raw: string & JSONable; // how to mark $ReadOnly with Flow?
+  _comparisonArray: number[];
+  _loggingArray: number[] & JSONable;
```

Comment:
Do these `& JSONable` have any effect? I think `string` and `number[]` should be subtypes of `JSONable` already.

Reply:
(And I don't think it's helpful to the human reader to redundantly say that here: it's a basic fact of JSON that it cleanly handles strings, numbers, and arrays of them.)

Reply:
True. I had left a code comment explaining that this was done for the reader's benefit, to whitelist the fields that could dependably be sent to Sentry, but this is better done in the JSDoc.